syslog-ng: 3.6.2 -> 3.9.1

basvandijk opened the pull request:
@fpletz this fixes https://github.com/NixOS/nixpkgs/issues/20153.

Since syslog-ng is currently broken on 16.09 this should be cherry-picked on `release-16.09`.

I tested it with the following configuration:

```nix
{
  services.syslog-ng = {
    enable = true;
    extraConfig = ''
      source s_network_tcp {
        network (
          ip("0.0.0.0")
          port(601)
          transport("tcp")
        );
      };
     
      destination d_devlog {
        unix-dgram(
          /dev/log
          flags(syslog-protocol)
        );
      };
     
      log {
        source(s_network_tcp);
        destination(d_devlog);
      };
    '';
  };
}
```

If I then log a message using:

```
logger --server=localhost --tcp "Hello World"
```

I see the following in my `journalctl`:

```
Feb 13 18:07:15 bas.van.dijk.lumi syslog-ng[13387]: 1 2017-02-13T18:07:15+01:00 localhost 1 - - - 2017-02-13T18:07:15.046465+01:00 bas.van.dijk.lumi bas.van.dijk - - [timeQuality tzKnown="1" isSynced="1" syncAccuracy="229387"] Hello World
```

basvandijk committed:
syslog-ng: 3.6.2 -> 3.9.1

This should fix: https://github.com/NixOS/nixpkgs/issues/20153

grahamc commented:
In what way is syslog-ng broken on 16.09?

basvandijk commented:
@grahamc see https://github.com/NixOS/nixpkgs/issues/20153.

abbradar commented on pkgs/tools/system/syslog-ng/default.nix:
```diff
@@ -1,17 +1,39 @@
-stdenv.mkDerivation rec {
-  name = "syslog-ng-${version}";
+stdenv.mkDerivation {
+  name = "syslog-ng-3.9.1";
```
Please leave `version`. It was discussed (admittedly a long time ago and it'd be difficult to find) that we might move to using it insttead of parsing `name` some day; it'd definitely be easier to move if most packages already contain it. Disclosure: I want it to happen very much as current way seems ugly to me ^_^

basvandijk replied:
Sure no problem.

fpletz committed:
libivykis: 0.40 -> 0.41

fpletz committed:
syslog-ng: refactor

fpletz committed:
syslogng_incubator: 141106 -> 0.5.0, mark as broken

This is the current release but neither this nor the current master
is compatible with syslog-ng 3.8+. Committed work on update and
refactor of expression to make it easier in the future to bump.

basvandijk commented:
@fpletz thanks for the refactoring. It looks good!

Since I'm now using a local override of syslog-ng I don't care that much anymore if this gets cherry-picked on 16.09. Waiting on 17.03 is fine by me.
